File: packages/go/luna/auth/auth_test.go
package auth_test

import (
	"testing"

	"github.com/eclipse-softworks/luna-sdk-go/luna/auth"
)

const validAPIKey = "lk_test_aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"

func TestAPIKeyAuth(t *testing.T) {
	t.Run("creates auth with valid key", func(t *testing.T) {
		a := auth.NewAPIKeyAuth(validAPIKey)
		if a == nil {
			t.Fatal("expected auth to be created")
		}
	})

	t.Run("panics on empty key", func(t *testing.T) {
		defer func() {
			if r := recover(); r == nil {
				t.Fatal("expected panic")
			}
		}()
		auth.NewAPIKeyAuth("")
	})

	t.Run("panics on invalid format", func(t *testing.T) {
		defer func() {
			if r := recover(); r == nil {
				t.Fatal("expected panic")
			}
		}()
		auth.NewAPIKeyAuth("invalid-key")
	})

	t.Run("returns correct headers", func(t *testing.T) {
		a := auth.NewAPIKeyAuth(validAPIKey)
		headers, err := a.GetHeaders()
		if err != nil {
			t.Fatalf("unexpected error: %v", err)
		}

		expected := map[string]string{"X-Luna-Api-Key": validAPIKey}
		for k, v := range expected {
			if headers[k] != v {
				t.Errorf("expected %s=%s, got %s", k, v, headers[k])
			}
		}
	})

	t.Run("does not need refresh", func(t *testing.T) {
		a := auth.NewAPIKeyAuth(validAPIKey)
		if a.NeedsRefresh() {
			t.Error("expected NeedsRefresh to be false")
		}
	})
}

func TestTokenAuth(t *testing.T) {
	t.Run("creates auth with access token", func(t *testing.T) {
		a := auth.NewTokenAuth("access-token", "refresh-token", nil)
		if a == nil {
			t.Fatal("expected auth to be created")
		}
	})

	t.Run("panics on empty access token", func(t *testing.T) {
		defer func() {
			if r := recover(); r == nil {
				t.Fatal("expected panic")
			}
		}()
		auth.NewTokenAuth("", "refresh-token", nil)
	})

	t.Run("returns correct headers", func(t *testing.T) {
		a := auth.NewTokenAuth("access-token", "refresh-token", nil)
		headers, err := a.GetHeaders()
		if err != nil {
			t.Fatalf("unexpected error: %v", err)
		}

		expected := "Bearer access-token"
		if headers["Authorization"] != expected {
			t.Errorf("expected Authorization=%s, got %s", expected, headers["Authorization"])
		}
	})

	t.Run("does not need refresh without expiry", func(t *testing.T) {
		a := auth.NewTokenAuth("access-token", "refresh-token", nil)
		if a.NeedsRefresh() {
			t.Error("expected NeedsRefresh to be false")
		}
	})
}

File: packages/typescript/tests/auth.test.ts
import { describe, it, expect } from 'vitest';
import { ApiKeyAuth } from '../src/auth/api-key.js';
import { TokenAuth } from '../src/auth/token.js';

describe('ApiKeyAuth', () => {
    const validApiKey = 'lk_test_12345678901234567890123456789012';

    it('should create auth with valid API key', () => {
        const auth = new ApiKeyAuth(validApiKey);
        expect(auth).toBeInstanceOf(ApiKeyAuth);
    });

    it('should throw error for empty API key', () => {
        expect(() => new ApiKeyAuth('')).toThrow('API key is required');
    });

    it('should throw error for invalid API key format', () => {
        expect(() => new ApiKeyAuth('invalid-key')).toThrow('Invalid API key format');
        expect(() => new ApiKeyAuth('lk_invalid_key')).toThrow('Invalid API key format');
        expect(() => new ApiKeyAuth('lk_test_short')).toThrow('Invalid API key format');
    });

    it('should return correct headers', async () => {
        const auth = new ApiKeyAuth(validApiKey);
        const headers = await auth.getHeaders();

        expect(headers).toEqual({
            'Authorization': `Bearer ${validApiKey}`,
        });
    });

    it('should not need refresh', () => {
        const auth = new ApiKeyAuth(validApiKey);
        expect(auth.needsRefresh()).toBe(false);
    });
});

describe('TokenAuth', () => {
    const accessToken = 'test-access-token';
    const refreshToken = 'test-refresh-token';

    it('should create auth with access token', () => {
        const auth = new TokenAuth({ accessToken });
        expect(auth).toBeInstanceOf(TokenAuth);
    });

    it('should throw error for empty access token', () => {
        expect(() => new TokenAuth({ accessToken: '' })).toThrow('Access token is required');
    });

    it('should return correct headers', async () => {
        const auth = new TokenAuth({ accessToken });
        const headers = await auth.getHeaders();

        expect(headers).toEqual({
            Authorization: `Bearer ${accessToken}`,
        });
    });

    it('should not need refresh without expiry', () => {
        const auth = new TokenAuth({ accessToken });
        expect(auth.needsRefresh()).toBe(false);
    });

    it('should update tokens', () => {
        const auth = new TokenAuth({ accessToken, refreshToken });

        auth.updateTokens({
            accessToken: 'new-access-token',
            refreshToken: 'new-refresh-token',
            expiresAt: new Date(Date.now() + 3600000),
        });

        // Token was updated (can verify via getHeaders)
        expect(auth.needsRefresh()).toBe(false);
    });
});
